src/radical/pilot/pmgr/launching/saga.py

```diff
@@ -0,0 +1,268 @@
+    #
+    def __init__(self, name, log, prof, state_cb):
+
+        PilotLauncherBase.__init__(self, name, log, prof, state_cb)
+
+        self._saga_jobs = dict()      # pid      : rs.Job
+        self._saga_js   = dict()      # resource : rs.JobService
+        self._saga_lock = ru.RLock()  # lock for above
+
+        # FIXME: get session from launching component
+        self._saga_session = rs.Session()
+
+        # FIXME: make interval configurable
+        # FIXME: run as daemon watcher thread
+        #
+      # self._watcher = mt.Thread(target=self._pilot_watcher)
+      # self._watcher.daemon = True
+      # self._watcher.start()
+
+
+    # --------------------------------------------------------------------------
+    #
+    def finalize(self):
+
+        # FIXME: terminate thread
+
+        with self._saga_lock:
+
+            # cancel pilots
+            for _, job in self._saga_jobs:
+                job.cancel()
+
+            # close job services
+            for url, js in self._saga_js.items():
+                self._log.debug('close js %s', url)
+                js.close()
+
+
+    # --------------------------------------------------------------------------
+    #
+    def can_launch(self, rcfg, pilot):
+
+        # SAGA can launch all pilots
+        return True
+
+
+    # --------------------------------------------------------------------------
+    #
+    def launch_pilots(self, rcfg, pilots):
+
+        js_ep  = rcfg['job_manager_endpoint']
+        with self._saga_lock:
+            if js_ep in self._saga_js:
+                js = self._saga_js[js_ep]
+            else:
+                js = rs.job.Service(js_ep, session=self._saga_session)
+                self._saga_js[js_ep] = js
+
+        # now that the scripts are in place and configured,
+        # we can launch the agent
+        jc = rs.job.Container()
+
+        for pilot in pilots:
+
+            jd_dict = pilot['jd_dict']
+
+            saga_jd_supplement = dict()
+            if 'saga_jd_supplement' in jd_dict:
+                saga_jd_supplement = jd_dict['saga_jd_supplement']
+                del(jd_dict['saga_jd_supplement'])
+
+
+            jd = rs.job.Description()
+            for k, v in jd_dict.items():
+                jd.set_attribute(k, v)
+
+            # set saga_jd_supplement if not already set
+            for key, val in saga_jd_supplement.items():
+                if not jd[key]:
+                    self._log.debug('supplement %s: %s', key, val)
+                    jd[key] = val
+
+            # set saga job description attribute based on env variable(s)
+            # FIXME: is this used?
+            if os.environ.get('RADICAL_SAGA_SMT'):
+                try:
+                    jd.system_architecture['smt'] = \
+                        int(os.environ['RADICAL_SAGA_SMT'])
+                except Exception as e:
+                    self._log.debug('SAGA SMT not set: %s' % e)
```

Comment:
we can remove this, since `smt` is set in base module

p.s. in base module we can either rename env variable (e.g., `RADICAL_SMT`) or even do not use at all (just use `smt` through config only)

Reply:
Ah, great - let me remove it then...